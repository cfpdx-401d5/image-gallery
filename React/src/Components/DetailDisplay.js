import React from 'react';

export default function DetailDisplay(props) {
    return (
        <div>
            <h4>{props.babyBunBun.title}</h4>
            <a href={props.babyBunBun.url}>See It Here!</a>
            <p>{props.babyBunBun.description}</p>
            <button onClick={() => props.deleteBunny(props.babyBunIndex, props.babyBunBun._id)}>Delete Bunny</button>
        </div>
    )
}

DetailDisplay.propTypes = {
    deleteBunny: React.PropTypes.func,
    babyBunIndex: React.PropTypes.number,
    babyBunBun: React.PropTypes.object
};