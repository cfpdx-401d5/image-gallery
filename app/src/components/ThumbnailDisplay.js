import React, { PropTypes } from 'react';

function ThumbnailDisplay(props) {
  return (
    <div className='image-divs'>
      <p>Title: {props.image.title} </p>
      <a href={props.image.url} > 
        <img className='thumbnail' alt='cute bunny' src={props.image.url} /> 
      </a>
      <button className='kill-button' onClick={() => props.deleteImage(props.image) }>
        Delete
      </button>      
    </div>
  );
}

ThumbnailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
  deleteImage: PropTypes.func.isRequired,
};

export default ThumbnailDisplay;
