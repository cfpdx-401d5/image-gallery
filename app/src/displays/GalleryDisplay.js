import React, { PropTypes } from 'react';

import GalleryDisplayBits from './GalleryDisplayBits';

function GalleryDisplay(props) {
  const imagesAsArticles = props.images.map(imageObj => {
    return <GalleryDisplayBits key={imageObj.id} title={imageObj.title} url={imageObj.url} description={imageObj.description} deleteImage={props.deleteImage} id={imageObj.id}/>;
  });

  return (
    <div>
        {imagesAsArticles}
    </div>
  );
}

GalleryDisplay.propTypes = {
  deleteImage: PropTypes.func.isRequired, 
  images: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  })).isRequired,
};


export default GalleryDisplay;
