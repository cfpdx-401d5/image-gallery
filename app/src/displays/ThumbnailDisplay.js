import React, { PropTypes } from 'react';

import ThumbnailDisplayBits from './ThumbnailDisplayBits';

function ThumbnailDisplay(props) {
  const imagesAsTableRows = props.images.map(imageObj => {
    return <ThumbnailDisplayBits key={imageObj.id} title={imageObj.title} url={imageObj.url} deleteImage={props.deleteImage} id={imageObj.id} />;
  });

  return (
    <table>
        <tbody>
            {imagesAsTableRows}
        </tbody>
    </table>
  );
}

ThumbnailDisplay.propTypes = {
  deleteImage: PropTypes.func.isRequired, 
  images: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  })).isRequired,
};


export default ThumbnailDisplay;
