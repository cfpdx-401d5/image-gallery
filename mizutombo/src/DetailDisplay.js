import React, { PropTypes } from 'react';

function DetailDisplay(props) {
  return (
    <div className='image-divs'>
      <p> Title: {props.image.title} </p>
      <p> Description: {props.image.description} </p>
      <a href={props.image.url}> Image Link </a>
      <button className='kill-button' onClick={() => props.deleteImage(props.image) }>
        Delete
      </button>
    </div>
  );
}

DetailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
  deleteImage: PropTypes.func.isRequired,
};

export default DetailDisplay;
