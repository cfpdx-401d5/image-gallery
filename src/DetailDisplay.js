import React from 'react';

export default function DetailDisplay(props) {
    return (
        <div>
            <h4>{props.babyBunBun.title}</h4>
            <a href={props.babyBunBun.url}>See It Here!</a>
            <p>{props.babyBunBun.description}</p>
            <button onClick={() => props.deleteBunny(props.babyBunIndex)}>Delete Bunny</button>
        </div>
    )
}

DetailDisplay.propTypes = {
    deleteBunny: React.PropTypes.func,
    babyBunIndex: React.PropTypes.number,
    key: React.PropTypes.oneOfType([
        React.PropTypes.string,
        React.PropTypes.number,
    ]),
    babyBunBun: React.PropTypes.object
};