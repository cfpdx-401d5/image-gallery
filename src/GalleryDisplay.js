import React from 'react';

export default class GalleryDisplay extends React.Component {
    constructor(props) {
    super(props);
    this.state = {
      showBun: true
    };
    this.showHandler = this.showHandler.bind(this);
  }

    showHandler(showBunValue) { //move all view logic to new class in refactor
    this.setState({
      showBun: showBunValue
    });
  }

  render() {
    if (this.state.showBun) {
        return (
            <div>
                <h4>{this.props.babyBunBun.title}</h4>
                <img alt={this.props.babyBunBun.title} src={this.props.babyBunBun.url}/><br />
                <button onClick={() => this.showHandler(false)}>Hide Bunny</button>
                <button onClick={() => this.props.deleteBunny(this.props.babyBunIndex)}>Delete Bunny</button>
                <p>{this.props.babyBunBun.description}</p>
            </div>
        )
    }
    else {
        return (
            <button onClick={() => this.showHandler(true)}>Show Bunny</button>
        )
    }
  }
}