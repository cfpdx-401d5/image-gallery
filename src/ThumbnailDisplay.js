import React, { PropTypes } from 'react';

function ThumbnailDisplay(props) {
  return (
    <div className='image-divs'>
      <p> Title: {props.image.title} </p>
      <img className='thumbnail' src={props.image.url} alt='cute bunny'/>
    </div>
  );
}

ThumbnailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
};

export default ThumbnailDisplay;
