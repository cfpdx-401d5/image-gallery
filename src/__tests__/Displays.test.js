import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ThumbnailDisplay from '../components/display/Thumbnail';
import DetailDisplay from '../components/display/Detail';
import GalleryDisplay from '../components/display/Gallery';

describe('DetailDisplay', () => {
    it('displays one detail resource', () => {
        let output = shallow(<DetailDisplay key={new Date()} resource={{ title: 'a', url: 'b', description: 'c' }} onDelete={() => { } } />);
        expect(toJson(output)).toMatchSnapshot();
    });
});

describe('ThumbnailDisplay', () => {
    it('displays one thumbnail resource', () => {
        let output = shallow(<ThumbnailDisplay key={new Date()} resource={{ title: 'a', url: 'b', description: 'c' }} onDelete={() => { } } />);
        expect(toJson(output)).toMatchSnapshot();
    });
});

describe('GalleryDisplay', () => {
    it('displays one gallery resource', () => {
        let output = shallow(<GalleryDisplay key={new Date()} resource={{ title: 'a', url: 'b', description: 'c' }} onDelete={() => { } } />);
        expect(toJson(output)).toMatchSnapshot();
    });
});
