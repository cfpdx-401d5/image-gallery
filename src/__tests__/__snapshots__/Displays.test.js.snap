exports[`DetailDisplay displays one detail resource 1`] = `
<div>
  <ul>
    <li>
      Title: 
      a
    </li>
    <li>
      Url: 
      b
    </li>
    <li>
      Description: 
      c
    </li>
  </ul>
  <button
    onClick={[Function]}>
    delete
  </button>
</div>
`;

exports[`GalleryDisplay displays one gallery resource 1`] = `
<div>
  <h1>
    Title: 
    a
  </h1>
  <img
    alt="Cute Bunny"
    className="gallery"
    src="b" />
  <p>
    Description: 
    c
  </p>
  <button
    onClick={[Function]}>
    delete
  </button>
</div>
`;

exports[`ThumbnailDisplay displays one thumbnail resource 1`] = `
<div>
  <h1>
    Title: 
    a
  </h1>
  <img
    alt="Cute Bunny"
    className="thumbnail"
    src="b" />
  <button
    onClick={[Function]}>
    delete
  </button>
</div>
`;
